dependencies.yaml

```
@@ -32,36 +32,64 @@ dependencies:
           - c-compiler
           - cxx-compiler
           - make
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+              cuda: "11.8"
+            packages:
+              - nvcc_linux-64=11.8
+          - matrix:
+              arch: aarch64
+              cuda: "11.8"
+            packages:
+              - nvcc_linux-aarch64=11.8
+          - matrix:
+              cuda: "12.0"
+            packages:
+              - cuda-version=12.0
+              - cuda-nvcc
   cudatoolkit:
     specific:
       - output_types: conda
         matrices:
           - matrix:
               cuda: "11.2"
             packages:
-              - cudatoolkit=11.2
+              - cuda-version=11.2
+              - cudatoolkit
               - gcc<11.0.0
               - sysroot_linux-64==2.17
           - matrix:
               cuda: "11.4"
             packages:
-              - cudatoolkit=11.4
+              - cuda-version=11.4
+              - cudatoolkit
               - gcc<11.0.0
               - sysroot_linux-64==2.17
           - matrix:
               cuda: "11.5"
             packages:
-              - cudatoolkit=11.5
+              - cuda-version=11.5
+              - cudatoolkit
               - sysroot_linux-64==2.17
           - matrix:
               cuda: "11.6"
             packages:
-              - cudatoolkit=11.6
+              - cuda-version=11.6
+              - cudatoolkit
               - sysroot_linux-64==2.17
           - matrix:
               cuda: "11.8"
             packages:
-              - cudatoolkit=11.8
+              - cuda-version=11.8
+              - cudatoolkit
```

This may need to be applied to CUDA 11.5 and 11.6 above, too. I'm not sure when GCC 11 support was added.
```suggestion
              - cudatoolkit
              - gcc<12.0.0
```

---

CUDA 11.4 introduced support for gcc-11.

---

Constrained the gcc version for all the CUDA versions